File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/spi/IotPropertySpecProvider.java
package site.addzero.biz.spec.iot.spi;

import site.addzero.biz.spec.iot.IotPropertySpec;
import site.addzero.biz.spec.iot.IotThingRef;

import java.util.List;

/**
 * Consumer-owned thing model provider.
 */
public interface IotPropertySpecProvider {

    String getName();

    boolean supports(IotThingRef thingRef);

    List<IotPropertySpec> getPropertySpecs(IotThingRef thingRef);
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/spi/IotPropertySpecProvider.kt
package site.addzero.biz.spec.iot.spi

import site.addzero.biz.spec.iot.IotPropertySpec
import site.addzero.biz.spec.iot.IotThingRef

/**
 * Consumer-owned thing model provider.
 */
interface IotPropertySpecProvider {

    val name: String

    fun supports(thingRef: IotThingRef): Boolean

    fun getPropertySpecs(thingRef: IotThingRef): List<IotPropertySpec>
}
